### DISCUSSION TITLE
Vercel-Postgres problem with sanitizing parameters

### DISCUSSION THREAD

**Discussion by DenisBytes** [vercel/next.js#59974] [Help]
### Summary

const optionsLearningSchema = z.object({
    user_id: z.string(),
    chapter: z.coerce.number()  
});
export async function updateOptionLearning(prevState: any, formData: FormData) {
    try{
        const {user_id, chapter} = optionsLearningSchema.parse({
            user_id: formData.get('user_id'),
            chapter: formData.get('chapter'),
        });
        const wow = sql`UPDATE options_learning SET chapter${chapter} = TRUE WHERE user_id = ${user_id}`;
        const wowResult = await wow;
        console.log('Updated options learning');
    }catch(error){
        console.error('Failed to update options learning:', error);
        throw new Error('Failed to update options learning.');
    }
}

In the code above im trying to update the field "chapter1" but the problem is that the "sql" function(?) is translating the parameter (numbr) chapter (1) to "chapter$1". with strings the same thing: "chapter'1'".

is there a way to get around this?

**Comment by matthewraaff** [comment]
Hey,

Instead of building the sql query like that, you could use parameterised queries, which can help prevent sql injection and fix your problem. [Here's](https://www.sqlshack.com/using-parameterized-queries-to-avoid-sql-injection/) a really good resource for learning more about it - please have a read before putting your code into production so that you ship secure code using sql.

For example you could try something like this.

```js
export async function updateOptionLearning(prevState: any, formData: FormData) {
  try {
    const { user_id, chapter } = optionsLearningSchema.parse({
      user_id: formData.get('user_id'),
      chapter: formData.get('chapter'),
    });

    const query = 'UPDATE options_learning SET chapter$1 = TRUE WHERE user_id = $2';
    const wowResult = await sql.query(query, [chapter, user_id]);

    console.log('Updated options learning');
  } catch (error) {
    console.error('Failed to update options learning:', error);
    throw new Error('Failed to update options learning.');
  }
}
```

It should avoid sql injection, and fix the input for your sql query. hope this helps :)
